modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/internal/search/spi/model/query/contributor/CPDefinitionModelPreFilterContributor.java

```diff
@@ -48,26 +57,236 @@ public void contribute(
 		BooleanFilter booleanFilter, ModelSearchSettings modelSearchSettings,
 		SearchContext searchContext) {
 
+		booleanFilter.addRequiredTerm(Field.HIDDEN, false);
+
+		_filterByBasePrice(booleanFilter, searchContext);
+		_filterByDefinitionLinks(booleanFilter, searchContext);
+		_filterByExcludedCPDefinitionId(booleanFilter, searchContext);
+		_filterByPublished(booleanFilter, searchContext);
+		_filterByStatuses(booleanFilter, searchContext);
+		_filterBySubscriptionEnabled(booleanFilter, searchContext);
```

Comment:
https://github.com/liferay-commerce/liferay-portal/pull/4003#discussion_r1261142990

Reply:
Direct port of `postProcessContextBooleanFilter`: https://github.com/liferay-commerce/liferay-portal/blob/master/modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/internal/search/CPDefinitionIndexer.java#L124